google-pubsub/1.0/modules/ROOT/pages/google-pubsub-connector-reference.adoc

```diff
@@ -0,0 +1,641 @@
+* PUBSUB:INVALID_ARGUMENT
+* PUBSUB:MAX_OUTSTANDING_BATCH_SIZE_REACHED
+* PUBSUB:MAX_OUTSTANDING_ELEMENT_COUNT_REACHED
+* PUBSUB:NOT_FOUND
+* PUBSUB:PERMISSION_DENIED
+* PUBSUB:RETRY_EXHAUSTED
+* PUBSUB:UNAVAILABLE
+
+
+[[SeekMessages]]
+== Seek Messages
+`<pubsub:seek-messages>`
+
+
+Supports bulk acknowledging or un-acknowledging of messages to a given snapshot or point of time based on the seekTarget provided. Seeking to a time marks every message received by Pub/Sub before the time as acknowledged, and all messages received after the time as unacknowledged. You can replay and reprocess previously acknowledged messages when seeking to a time in the past or purge messages when seeking to a time in the future.
+
+
+If you seek to a timestamp using a subscription with a filter, the Pub/Sub service redelivers only the messages in the timestamp that match the filter of the subscription making the seek request. The snapshot feature enables you to capture the message acknowledgment state of a subscription. Once a snapshot is created, it retains:
+
+* all messages that were unacknowledged in the source subscription at the time of the snapshot's creation.
+* any messages published to the topic thereafter. You can replay these unacknowledged messages by using a snapshot to seek to any of the topic's subscriptions.
+
+
+If you seek to a snapshot using a subscription with a filter, the Pub/Sub service redelivers only the messages in the snapshot that match the filter of the subscription making the seek request.
+
+
+=== Parameters
+
+[%header%autowidth.spread]
+|===
+| Name | Type | Description | Default Value | Required
+| Configuration | String | Name of the configuration to use. | | x
+| Config Ref a| ConfigurationProvider |  Name of the configuration used to execute this component. |  | x
+| Project ID a| String |  Globally unique identifier for your project. |  |
+| Subscription Name a| String |  Project-wide unique identifier for the subscription. |  |
+| Snapshot Name a| String |  Snapshot to seek to. The snapshot's topic must be the same as that of the provided subscription. |  |
+| Timestamp a| DateTime |  Specific time to seek to. Messages retained in the subscription that were published before this time are marked as acknowledged, and messages retained in the subscription that were published after this time are marked as unacknowledged.
+
+Note that this operation affects only messages retained in the subscription (configured by the combination of *messageRetentionDuration* and *retainAckedMessages*), for example, if the time corresponds to a point before the message retention window (or to a point before the system's notion of the subscription creation time), only retained messages are marked as unacknowledged, and already-expunged messages will not be restored. A timestamp is in RFC3339 UTC Zulu format, with nanosecond resolution and up to nine fractional digits, such as 2014-10-02T15:01:23Z and 2014-10-02T15:01:23.045123456Z. |  |
```

Comment:
If possible, use  *Message Retention Duration* and *Retain Acked Messages* if these are field names in the connector config.

Reply:
```suggestion
This operation affects only messages retained in the subscription (configured by the combination of *messageRetentionDuration* and *retainAckedMessages*). For example, if the time corresponds to a point before the message retention window or to a point before the subscription creation time, only retained messages are marked as unacknowledged and already-expunged messages are not restored. A timestamp is in RFC3339 UTC Zulu format, with nanosecond resolution and up to nine fractional digits, such as 2014-10-02T15:01:23Z and 2014-10-02T15:01:23.045123456Z. |  |
```